packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -44,3 +44,92 @@ const intervalOptions = [
 { value: 5, description: 'Yearly', name: 5,
 ];
 */
+
+let uncategorizedCategory = {
+  name: 'Uncategorized',
+  id: null,
+  uncat_id: '1',
+  hidden: 0,
+  offBudget: false,
+  transfer: false,
+  category: false,
+};
+let transferCategory = {
+  name: 'Transfers',
+  id: null,
+  uncat_id: '2',
+  hidden: 0,
+  transfer: true,
+  offBudget: false,
+  category: false,
+};
+let offBudgetCategory = {
+  name: 'Off Budget',
+  id: null,
+  uncat_id: '3',
+  hidden: 0,
+  offBudget: true,
+  transfer: false,
+  category: true,
+};
+
+let uncategouncatGrouprizedGroup = {
+  name: 'Uncategorized & Off Budget',
+  id: null,
+  hidden: 0,
+  categories: [uncategorizedCategory, transferCategory, offBudgetCategory],
+};
+
+export const categoryLists = (uncat, categories) => {
+  let categoryList = uncat
+    ? [
+        ...categories.list,
+        uncategorizedCategory,
+        transferCategory,
+        offBudgetCategory,
+      ]
+    : categories.list;
+  let categoryGroup = uncat
+    ? [...categories.grouped, uncategouncatGrouprizedGroup]
+    : categories.grouped;
+  return [categoryList, categoryGroup];
+};
+
+export const groupBySelections = (
+  groupBy,
+  categoryList,
+  categoryGroup,
+  payees,
+  accounts,
+) => {
+  let groupByList;
+  let groupByLabel;
+  switch (groupBy) {
+    case 'Category':
+      groupByList = categoryList;
+      groupByLabel = 'category';
+      break;
+    case 'Group':
+      groupByList = categoryGroup;
+      groupByLabel = 'categoryGroup';
+      break;
+    case 'Payee':
+      groupByList = payees;
+      groupByLabel = 'payee';
+      break;
+    case 'Account':
+      groupByList = accounts;
+      groupByLabel = 'account';
+      break;
+    case 'Month':
+      groupByList = categoryList;
+      groupByLabel = 'category';
+      break;
+    case 'Year':
+      groupByList = categoryList;
+      groupByLabel = 'category';
+      break;
+    default:
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏would be good to add a fatal error here to stop program execution if this condition is ever met (but I hope it is never met).
```suggestion
    default:
        throw new Error('Something');
```